File: WinPaletter/ColorItemControllers/ColorInfoDragDrop.vb
Imports WinPaletter.NativeMethods
Imports WinPaletter.XenonCore

Public Class ColorInfoDragDrop

#Region "Form Shadow"

    Private aeroEnabled As Boolean

    Protected Overrides ReadOnly Property CreateParams() As CreateParams
        Get
            CheckAeroEnabled()
            Dim cp As CreateParams = MyBase.CreateParams
            If Not aeroEnabled Then
                cp.ClassStyle = cp.ClassStyle Or Dwmapi.CS_DROPSHADOW
                cp.ExStyle = cp.ExStyle Or 33554432
                Return cp
            Else
                Return cp
            End If
        End Get
    End Property

    Protected Overrides Sub WndProc(ByRef m As Message)
        Select Case m.Msg
            Case Dwmapi.WM_NCPAINT
                Dim val = 2
                If aeroEnabled Then
                    Dwmapi.DwmSetWindowAttribute(Handle, If(GetRoundedCorners(), 2, 1), val, 4)
                    Dim bla As New Dwmapi.MARGINS()
                    With bla
                        .bottomHeight = 1
                        .leftWidth = 1
                        .rightWidth = 1
                        .topHeight = 1
                    End With
                    Dwmapi.DwmExtendFrameIntoClientArea(Handle, bla)
                End If
                Exit Select
        End Select

        MyBase.WndProc(m)
    End Sub

    Private Sub CheckAeroEnabled()
        If Environment.OSVersion.Version.Major >= 6 Then
            Dim Com As Boolean
            Dwmapi.DwmIsCompositionEnabled(Com)
            aeroEnabled = Com
        Else
            aeroEnabled = False
        End If
    End Sub
#End Region

    ReadOnly _dark As Single = 0.7

    Private Sub ColorInfoDragDrop_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Location = MousePosition + New Point(15, 15)

        LoadLanguage
        ApplyDarkMode(Me)

        Label6.Font = My.Application.ConsoleFontMedium
        Label7.Font = My.Application.ConsoleFontMedium
        Label8.Font = My.Application.ConsoleFontMedium
        Label9.Font = My.Application.ConsoleFontMedium
        Label10.Font = My.Application.ConsoleFontMedium
        Label11.Font = My.Application.ConsoleFontMedium
        Label12.Font = My.Application.ConsoleFontMedium
        Label13.Font = My.Application.ConsoleFontMedium
    End Sub

    Private Sub Color_From_BackColorChanged(sender As Object, e As EventArgs) Handles Color_From.BackColorChanged
        Dim Color As Color = CType(sender, Panel).BackColor
        XenonAnimatedBox1.Color = Color
        XenonAnimatedBox1.Color1 = Color

        Label6.Text = Color.ReturnFormat(ColorFormat.RGB, True, Color.A < 255).Replace(" ", ", ")
        Label7.Text = Color.ReturnFormat(ColorFormat.HEX, True, Color.A < 255).Replace(" ", ", ")
        Label8.Text = Color.ReturnFormat(ColorFormat.HSL, True, Color.A < 255).Replace(" ", ", ")
        Label9.Text = Color.ReturnFormat(ColorFormat.Dec, True, Color.A < 255).Replace(" ", ", ")

        Label6.ForeColor = If(Color.IsDark, Color.White, Color.Black)
        Label7.ForeColor = Label6.ForeColor
        Label8.ForeColor = Label6.ForeColor
        Label9.ForeColor = Label6.ForeColor
    End Sub

    Private Sub Color_To_BackColorChanged(sender As Object, e As EventArgs) Handles Color_To.BackColorChanged
        Dim Color As Color = CType(sender, Panel).BackColor
        XenonAnimatedBox1.Color2 = Color

        Label13.Text = Color.ReturnFormat(ColorFormat.RGB, True, Color.A < 255).Replace(" ", ", ")
        Label12.Text = Color.ReturnFormat(ColorFormat.HEX, True, Color.A < 255).Replace(" ", ", ")
        Label11.Text = Color.ReturnFormat(ColorFormat.HSL, True, Color.A < 255).Replace(" ", ", ")
        Label10.Text = Color.ReturnFormat(ColorFormat.Dec, True, Color.A < 255).Replace(" ", ", ")

        Label10.ForeColor = If(Color.IsDark, Color.White, Color.Black)
        Label11.ForeColor = Label10.ForeColor
        Label12.ForeColor = Label10.ForeColor
        Label13.ForeColor = Label10.ForeColor
    End Sub
End Class

File: WinPaletter/ColorItemControllers/ColorInfoDragDrop.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using WinPaletter.NativeMethods;

namespace WinPaletter
{

    public partial class ColorInfoDragDrop
    {

        #region Form Shadow

        protected override CreateParams CreateParams
        {
            get
            {
                var cp = base.CreateParams;
                if (!DWMAPI.IsCompositionEnabled())
                {
                    cp.ClassStyle = cp.ClassStyle | DWMAPI.CS_DROPSHADOW;
                    cp.ExStyle = cp.ExStyle | 33554432;
                    return cp;
                }
                else
                {
                    return cp;
                }
            }
        }

        public ColorInfoDragDrop()
        {
            InitializeComponent();
        }

        protected override void WndProc(ref Message m)
        {
            switch (m.Msg)
            {
                case DWMAPI.WM_NCPAINT:
                    {
                        int val = 2;
                        if (DWMAPI.IsCompositionEnabled())
                        {
                            DWMAPI.DwmSetWindowAttribute(Handle, Program.Style.RoundedCorners ? 2 : 1, ref val, 4);
                            DWMAPI.MARGINS bla = new();
                            {
                                bla.bottomHeight = 1;
                                bla.leftWidth = 1;
                                bla.rightWidth = 1;
                                bla.topHeight = 1;
                            }
                            DWMAPI.DwmExtendFrameIntoClientArea(Handle, ref bla);
                        }
                        break;
                    }
            }

            base.WndProc(ref m);
        }
        #endregion

        private readonly float _dark = 0.7f;

        private void ColorInfoDragDrop_Load(object sender, EventArgs e)
        {
            Location = MousePosition + (Size)new Point(15, 15);

            this.LoadLanguage();
            ApplyStyle(this);

            Label6.Font = Fonts.ConsoleMedium;
            Label7.Font = Fonts.ConsoleMedium;
            Label8.Font = Fonts.ConsoleMedium;
            Label9.Font = Fonts.ConsoleMedium;
            Label10.Font = Fonts.ConsoleMedium;
            Label11.Font = Fonts.ConsoleMedium;
            Label12.Font = Fonts.ConsoleMedium;
            Label13.Font = Fonts.ConsoleMedium;
        }

        private void Color_From_BackColorChanged(object sender, EventArgs e)
        {
            var Color = ((Panel)sender).BackColor;
            BackColor = Program.Style.DarkMode ? Color.Dark(_dark) : Color.LightLight();

            Label6.Text = Color.ReturnFormat(ColorsExtensions.ColorFormat.RGB, true, Color.A < 255).Replace(" ", ", ");
            Label7.Text = Color.ReturnFormat(ColorsExtensions.ColorFormat.HEX, true, Color.A < 255).Replace(" ", ", ");
            Label8.Text = Color.ReturnFormat(ColorsExtensions.ColorFormat.HSL, true, Color.A < 255).Replace(" ", ", ");
            Label9.Text = Color.ReturnFormat(ColorsExtensions.ColorFormat.Dec, true, Color.A < 255).Replace(" ", ", ");

            Label6.ForeColor = Color.IsDark() ? System.Drawing.Color.White : System.Drawing.Color.Black;
            Label7.ForeColor = Label6.ForeColor;
            Label8.ForeColor = Label6.ForeColor;
            Label9.ForeColor = Label6.ForeColor;
        }

        private void Color_To_BackColorChanged(object sender, EventArgs e)
        {
            var Color = ((Panel)sender).BackColor;

            Label13.Text = Color.ReturnFormat(ColorsExtensions.ColorFormat.RGB, true, Color.A < 255).Replace(" ", ", ");
            Label12.Text = Color.ReturnFormat(ColorsExtensions.ColorFormat.HEX, true, Color.A < 255).Replace(" ", ", ");
            Label11.Text = Color.ReturnFormat(ColorsExtensions.ColorFormat.HSL, true, Color.A < 255).Replace(" ", ", ");
            Label10.Text = Color.ReturnFormat(ColorsExtensions.ColorFormat.Dec, true, Color.A < 255).Replace(" ", ", ");

            Label10.ForeColor = Color.IsDark() ? System.Drawing.Color.White : System.Drawing.Color.Black;
            Label11.ForeColor = Label10.ForeColor;
            Label12.ForeColor = Label10.ForeColor;
            Label13.ForeColor = Label10.ForeColor;
        }
    }
}